waspc/cli/src/Wasp/Cli/Command/News/Listing.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Wasp.Cli.Command.News.Listing
+  ( -- * News Listing
+
+    -- This modules captures all decisions Wasp has to make when showing Wasp
+    -- news under different contexts (i.e., who requested the news listing):
+    --   - Which news to show under which context.
+    --   - Which news to mark as seen.
+    --   - Whether the user should interactively confirm they've read the news.
+    --   - etc.
+    --
+    -- We tried to caputure most of the logic in pure function to simplify
+    -- testing.
+    NewsListing (..),
+    getNewsToShow,
+    isConfirmationRequired,
+    getNewsToMarkAsSeen,
+    processNewsListing,
+    shouldWaspListMustSeeNews,
+    getNewsFromListing,
+  )
+where
+
+import Control.Monad (unless, when)
+import Data.List (intercalate)
+import qualified Data.Time as T
+import Wasp.Cli.Command.News.Core (NewsEntry (..), NewsLevel (..))
+import Wasp.Cli.Command.News.Display (showNewsEntry)
+import Wasp.Cli.Command.News.LocalNewsState
+  ( LocalNewsState,
+    emptyLocalNewsState,
+    markNewsAsSeen,
+    saveLocalNewsState,
+    setLastListingTimestamp,
+    wasLastLisingMoreThanNHoursAgo,
+    wasNewsEntrySeen,
+  )
+import Wasp.Cli.Interactive (askForConfirmationWithTimeout)
+import Wasp.Util.Terminal (styleCode)
+
+shouldWaspListMustSeeNews :: LocalNewsState -> IO Bool
+shouldWaspListMustSeeNews = wasLastLisingMoreThanNHoursAgo 24
+
+data NewsListing
+  = UserRequestedAllNews [NewsEntry]
+  | WaspRequestedMustSeeNews [NewsEntry]
+  deriving (Show, Eq)
+
+getNewsToShow :: LocalNewsState -> NewsListing -> [NewsEntry]
+getNewsToShow localState = \case
+  UserRequestedAllNews allNews -> allNews
+  WaspRequestedMustSeeNews allNews
+    | doesUserHaveNewsHistory localState -> unseenNewsThatUserMustSee allNews
+    | otherwise -> []
+  where
+    unseenNewsThatUserMustSee = filter ((>= Important) . level) . filter isUnseen
+    isUnseen = not . wasNewsEntrySeen localState
+
+isConfirmationRequired :: LocalNewsState -> NewsListing -> Bool
+isConfirmationRequired localState listing = case listing of
+  UserRequestedAllNews _ -> False
+  WaspRequestedMustSeeNews _ -> any ((== Critical) . level) $ getNewsToShow localState listing
+
+getNewsToMarkAsSeen :: LocalNewsState -> NewsListing -> [NewsEntry]
+getNewsToMarkAsSeen localState listing = case listing of
+  UserRequestedAllNews allNews -> allNews
+  WaspRequestedMustSeeNews allNews
+    | not $ doesUserHaveNewsHistory localState -> allNews
+    | isConfirmationRequired localState listing -> getNewsToShow localState listing
+    | otherwise -> []
+
+processNewsListing :: LocalNewsState -> NewsListing -> IO ()
+processNewsListing localState listing = do
```

Comment:
The new behavior is:
  - When Wasp shows critical news during `wasp start`, wait for confirmation but allow an answer other than "y" and time out after 10 seconds. If the confirmation times out or isn't "y," the news are shown again in a day.
  - When Wasp shows important news during wasp start, tell the users they have to run `wasp news` to mark them as read. @infomiho I didn't do the tiemout because it felt to similar to how we behave when there are critical news.
  - When the user runs `wasp news` themselves, they get all news and no other confirmation/messages